### PR TITLE
Milestone 4 logic

### PR DESCRIPTION
**Pull request by peterrobert**
In this milestone, I created the following:

1. Calculate.js  file for the logic.
2. Operate.js file for the logic.

### CODE REVIEW

File: src/components/Button.js [hunk 1]
```diff
@@ -21,4 +20,6 @@ export const Button = (props) => {
 
 Button.propTypes = {
   name: PropTypes.string.isRequired,
+  color: PropTypes.string.isRequired,
```
Comment (mosaaleb) [inline src/components/Button.js]: - [x] I got a console error here:
Screenshot from 2021-01-14 22-15-38

Make sure to pass the color prop or declare a default prop [declaring default props](https://reactjs.org/docs/react-without-es6.html#declaring-default-props)

File: src/components/Button.js [hunk 2]
```diff
@@ -1,15 +1,14 @@
-import { React } from "react";
-import PropTypes from "prop-types";
+import { React } from 'react';
+import PropTypes from 'prop-types';
 
-export const Button = (props) => {
+export const Button = props => {
```
Comment (mosaaleb) [inline src/components/Button.js]: - [x] **[OPTIONAL]** We can directly de-structure objects inside function arguments, which will save use a few lines.

```js
export const Button = ({ name, color, wide}) => {
```

File: src/components/Button.js [hunk 3]
```diff
@@ -21,4 +20,6 @@ export const Button = (props) => {
 
 Button.propTypes = {
   name: PropTypes.string.isRequired,
+  color: PropTypes.string.isRequired,
+  wide: PropTypes.string.isRequired,
```
Comment (mosaaleb) [inline src/components/Button.js]: - [x] I got also another console error here:
Screenshot from 2021-01-14 22-23-47

Let's fix this.

File: src/logic/operate.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    result = first.plus(second);
+    return result;
+  }
+
+  if (operation === 'X') {
+    result = first.times(second);
+    return result;
+  }
+
+  if (operation === '=') {
```
Comment (mosaaleb) [inline src/logic/operate.js]: - [x] We need to replace this condition with `%` which turns a number to precentage notation.

File: src/logic/calculate.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+/* eslint-disable no-unused-vars */
+import { operate } from './operate';
+
+export const calculate = (calculator, button) => {
+  let { total, next, operation } = calculator;
+
+  if (button === '+/-') {
```
Comment (mosaaleb) [inline src/logic/calculate.js]: - [x] Good job done here, but we need to handle more cases to cover all the possible scenarios.
Here are some cases to consider to structure the calculator object `total, next, operation`:
- what if the button is a number?
- what if the button is a number and the previous click is an operation?
- what if the button is '.'

Kindly make sure to cover all the possible scenarios.

File: .eslintrc.json [hunk 6]
```diff
@@ -1,3 +1,29 @@
...
+        "sourceType": "module"
+      },
+      "extends": ["airbnb", "plugin:react/recommended"],
+      "plugins": ["react"],
+      "rules": {
+        "react/jsx-filename-extension": ["warn", { "extensions": [".js", ".jsx"] }],
+        "react/react-in-jsx-scope": "off",
+        "import/no-unresolved": "off",
+        "no-shadow": "off",
+        "import/prefer-default-export": "off",
```
Comment (mosaaleb) [inline .eslintrc.json]: - [x] Linters config: Do not make any changes in config files - they represent style guidelines that you share with your team

**Review by mosaaleb — CHANGES_REQUESTED** [review]
#### Changes required

Well done @peterrobert :clap:. You have already implemented most of the requirements from this milestone. However, there are some console error and other requirements we need to handle before moving on. Kindly take a look at my comments below:

- [x] `node_modules` is currently tracked by git. Kindly make sure to properly add node_modules in .gitignore. Hint: remove the `\`. Read more about [gitignore rules](https://git-scm.com/docs/gitignore)

- [x] Let's import `calculate.js` in `App` component. 

Please handle these requirements and submit a new code review.

##### Muhammad Ebeid

File: src/logic/operate.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+      data = num1.times(num2);
+      break;
+    case '+':
+      data = num1.plus(num2);
+      break;
+    case '-':
+      data = num1.minus(num2);
+      break;
+    case '%':
+      data = num1.times(Big(0.01));
```
Comment (bellom) [inline src/logic/operate.js]: When it comes to `%` we only need one operand.
example: click `10` then `%` = 0.01
This should work.
Example: `num1 = (num1 *= 0.01).toString();`

You might want to add `toString()` at the end of your cases.

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @peterrobert  :clap: 

**Great work making `almost` all the necessary corrections from the previous review.**

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

**You seems to introduce a bug, `npm start` not working**
For some reason,  you pushed the `node_modules` folder to GitHub, you could ignore this in the `gitignore` file, perhaps that would help.

Try to delete the `node_modules`, add it to the `gitignore` and run `npm i`

Screenshot from your app;
image


image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/


### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi @peterrobert 👋

**You've done quite well here fixing most issues raised in the previous review.👌**

 However to move forward let's fix this:

- [x]  Delete node_modules from the development branch. To do this switch to the development branch then run `rm -rf node_modules` in your terminal then next push directly to the development branch. It's important that node_modules is removed from the development branch. Kindly fix this.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix this issue and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

**Review by mikenath223 — APPROVED** [review]
### CONGRATULATIONS  👏👏👏

This project meets the requirements for approval.

**STATUS: APPROVED ✅**


Don't forget to fill the project completion form👍
Good luck with the next project.

HAPPY CODING👨‍💻

**Pull request merged**